Jenkinsfile

```diff
@@ -59,6 +59,7 @@ pipeline {
         withDockerRegistry([credentialsId: "${env.JENKINS_DOCKER_CRED_ID}", url: ""]) {
           dir("$DIRECTORY/images/nginx") {
             sh "make push"
+            sh 'echo ${REGISTRY}/${IMAGE_NAME}:${TAG} >> images.list'
```

Comment:
To be safe on replays or rebuilds, you should add an `rm -f images.list` in the setup stage

Reply:
If Jenkins doesn't run the `cleanup` stage, that sounds like a bug in Jenkins not something that needs to be changed here. Are the special cases you mention documented by Jenkins? Should we make a bug report?

Reply:
Fixed in c114bfe